lnwallet/test/test_interface.go

```diff
@@ -1311,13 +1311,66 @@ func testListTransactionDetails(miner *rpctest.Harness,
 
 	// Query for transactions only in the latest block. We do not expect
 	// any transactions to be returned.
-	txDetails, err = alice.ListTransactionDetails(
+	txDetails, _, _, err = alice.ListTransactionDetails(
 		chainTip, chainTip, "", 0, 1000,
 	)
 	require.NoError(t, err, "unexpected error")
 	if len(txDetails) != 0 {
 		t.Fatalf("expected 0 transactions, got: %v", len(txDetails))
 	}
+
+	// Next we try to query for only 5 transactions.
```

Comment:
seems a perfect fit for table-driven tests, if it's easier, we can copy the part where we create all the txns in `testListTransactionDetails`, create a new test `testListTransactionDetailsOffset`, and check all the cases.
- when there are no txns, a larger `index_offset` should give us nothing.
- when there are no txns, a larger `max_transactions` should give us nothing. 
- when there are no txns, larger `max_transactions` and `index_offset` should give us nothing. 
- when there are 10 txns, check `index_offset=0, max_transactions=-1` for the default case
- when there are 10 txns, check `index_offset=5, max_transactions=-1` for a within bound offset
- when there are 10 txns, check `index_offset=9, max_transactions=-1` for an edge offset
- when there are 10 txns, check `index_offset=10, max_transactions=-1` for an outbound offset
- when there are 10 txns, check `index_offset=0, max_transactions=1` for a within bound max num
- when there are 10 txns, check `index_offset=0, max_transactions=10` for an edge max num
- when there are 10 txns, check `index_offset=0, max_transactions=11` for an outbound max num

Reply:
I have created a new commit and added a new test as you suggested.